src/components/shared/formItems/formHelpers.tsx

```diff
@@ -0,0 +1,117 @@
+import { PayoutMod } from 'models/mods'
+import { fromPermyriad } from 'utils/formatNumber'
+import { constants, utils } from 'ethers'
+import { BigNumber } from '@ethersproject/bignumber'
+
+import {
+  stripCommas,
+  formatWad,
+  parseWad,
+  fromWad,
+  parsePermyriad,
+} from 'utils/formatNumber'
+import { amountSubFee, amountAddFee } from 'utils/math'
+
+export type ModalMode = 'Add' | 'Edit' | undefined
+
+// Get total percentages from a list of mods such as in Project->'Edit payouts'
+export function getTotalPercentage(mods: PayoutMod[] | undefined) {
+  return (
+    mods?.reduce(
+      (acc, curr) => acc + parseFloat(fromPermyriad(curr.percent ?? '0')),
+      0,
+    ) ?? 0
+  )
+}
+
+// Ensures value is greater than 0 and less than 100
+export function validatePercentage(percent: number | undefined) {
+  if (percent === undefined || percent === 0) return Promise.reject('Required')
+  else if (percent > 100) return Promise.reject('Invalid')
+  return Promise.resolve()
+}
+
+// Validates an eth address from a modal where the address
+// is being added to a set of addresses (such as payout or
+// reserved token receivers)
+// Ensures address is not null, 0, or the same as an already added address
+export function validateEthAddress(
+  address: string,
+  mods: PayoutMod[],
+  modalMode: ModalMode,
+  editingModIndex: number | undefined,
+) {
+  // If user edits an (already approved) address and doesn't change it, we accept
+  if (
+    modalMode === 'Edit' &&
+    address === mods[editingModIndex ?? 0]?.beneficiary
+  )
+    return Promise.resolve()
+  else if (!address || !utils.isAddress(address))
+    return Promise.reject('Address is required')
+  else if (address === constants.AddressZero)
+    return Promise.reject('Cannot use zero address')
+  else if (mods.some(mod => mod.beneficiary === address))
+    return Promise.reject('A payout for this address already exists')
+  else return Promise.resolve()
+}
+
+// Returns formatted 'funding target after fee'
+// given 'funding target'
+export const targetToTargetSubFeeFormatted = (
+  target: string,
+  fee: BigNumber | undefined,
+) => {
+  const newTargetSubFee = amountSubFee(parseWad(target ?? ''), fee)
+  return stripCommas(formatWad(newTargetSubFee, { decimals: 4 }) || '0') // formatWad returns formatted bigNum with commas, must remove
```

Comment:
we should be able to replace `stripCommas(formatWad(x))` with `fromWad(x)`

Reply:
Oh yep I've just fixed those up. Just wondering if we're happy showing any number of decimals places on the funding target inputs or if we want to round them at some point:

<img width="630" alt="Screen Shot 2022-01-08 at 10 05 22 am" src="https://user-images.githubusercontent.com/96150256/148622345-8a3c84ea-27f7-4973-974f-354bc3992ec9.png">

Reply:
@peripheralist Oh yep I missed those! Just wondering if we're happy showing any number of decimals places or if we want to round them at some point (Jango tells me we store funding targets to 16 d.p.'s so it's not inaccurate as it's shown here): 

<img width="630" alt="Screen Shot 2022-01-08 at 10 05 22 am" src="https://user-images.githubusercontent.com/96150256/148622614-cfef2f98-67fc-44b4-a21c-7d2969bd1a37.png">

Reply:
I'm down to just show the full amount for clarity! I imagine it will usually be more clean if users are inputting round numbers